commit 15b158db3ae11cb835f2eb8d2eb48e09d1a4af48
Author: Andreas Gruenbacher <agruen@gnu.org>
Date:   Mon Jul 15 19:10:02 2019 +0200

    Avoid invalid memory access in context format diffs
    
    * src/pch.c (another_hunk): Avoid invalid memory access in context format
    diffs.

diff --git a/src/pch.c b/src/pch.c
index a500ad9..cb54e03 100644
--- a/src/pch.c
+++ b/src/pch.c
@@ -1328,6 +1328,7 @@ another_hunk (enum diff difftype, bool rev)
 		  ptrn_prefix_context = context;
 		ptrn_suffix_context = context;
 		if (repl_beginning
+		    || p_end <= 0
 		    || (p_end
 			!= p_ptrn_lines + 1 + (p_Char[p_end - 1] == '\n')))
 		  {
